lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -316,6 +316,9 @@ def ensure(module, base, state, names):
     # and fail with a message about what they can't.
     failures = []
     allow_erasing = False
+    if module.params.get('update_cache') and state in ['installed', 'present', 'latest']:
```

Comment:
Why require the state?
apt and zypper allow `update_cache` without any state and I don't see how dnf is different in that regard.